### DISCUSSION TITLE
why does maxima have these most inconvenient bottlenecks for no reason

### DISCUSSION THREAD

**Discussion by zmth** [wxMaxima-developers/wxmaxima#1599] [General]
`kill(a(k))`
and maxima gives improper argumenta(k)  and there is absolutely NO reason for not allowing this and there could
NEVER be any source of confusion or ambiguity in not allowing this !  Also
`tk:(1+2*k+3*k^2+n)/(1-k+k^2-n^2)`
ak:part(tk,1);
gives as it should, the expression we want but if i do
ak(n,k):=part(tk,1)
It just repeats the definition form and NOT what one wants which is the same output as for ak:...
Again no reason as could never cause any confusion or ambiguity. Another unnecessary bottleneck.
(for a[1]:0 thru 2 do for a[2]:a[1] thru 3 do a[3]:2*a[1]+a[2]);
maxima gives [ is an unknown.....NOw i do the exact same
(for a:0 thru 2 do for b:a thru 3 do c:2*a+b);  
and maxima accepts. What is the reason for maxima giving errors when there is no need to do so and not accept these
statements when there could never be any source of confusion or reason for not accepting as given. All it does is just cause inconvenience for the input person.

**Comment by daute** [comment]
(The issue here is a Maxima issue, it has nothing to do with the graphical frontend wxMaxima).

`:=` defines a function. So maybe you want something like
`ak(n):=part(tk,n)`
then `ak(1)` would give either the first part (the numerator) or `ak(2)` would return the second part (the denominator) of the expression `tk`?
Does that do, what you expect?

**Reply by zmth** [reply]
Actually i wasn't talking about graphical front end etc. BUt anyway I seem to recall long ago like many or a few years ago one of the developers or whoever did give a logical answer as to why

(for a[1]:0 thru 2 do for a[2]:a[1] thru 3 do a[3]:2a[1]+a[2]);....

type of input would create a problem like ambiguity, inconsistency etc. and why it could not be used but i can't seem to locate that discussion now. Does anyone know or recall where i could find that post. NOt sure whether it was under bugs or discussion or elsewhere though
